src/libraries/System.Private.CoreLib/src/System/Threading/NativeRuntimeEventSource.PortableThreadPool.cs

```diff
@@ -134,6 +119,9 @@ private unsafe void WriteThreadEvent(int eventId, uint numExistingThreads)
             if (IsEnabled(EventLevel.Informational, Keywords.ThreadingKeyword))
```

Comment:
I may have missed it, but what ensures that `IsEnabled` returns true?

Reply:
Sorry, I don't think I quite got what you're asking here.. Do you mean what ensures that the DoCommand callback is received for the NativeRuntimeEventSource?

NativeRuntimeEventSource registers itself with the native runtime upon initialization, and it will receive callback from the runtime from the standard callback handler, just like how any other managed EventSources receive callbacks when being enabled.

Essentially the runtime "thinks" there are two providers -- the native provider whose callback is the EtwCallbackCommon function and the managed provider whose callback is the managed callback function in EventSource.